### PR TITLE
refactor: Fix alert rules for failed collector.

### PR DESCRIPTION
**Pull request by dashmage**
Relates to https://github.com/canonical/prometheus-hardware-exporter/pull/54

### CODE REVIEW

File: tests/functional/test_charm.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 from pytest_operator.plugin import OpsTest
 
 logger = logging.getLogger(__name__)
+pytestmark = pytest.mark.asyncio(scope="module")
```
Comment (rgildein) [inline tests/functional/test_charm.py:17]: If this is needed due some bug, please add note here.

**Review by rgildein — COMMENTED** [review]
need informations

File: tests/functional/test_charm.py (hunk 1 above)
Comment (Pjack) [reply tests/functional/test_charm.py:17]: +1

**Review by Pjack — APPROVED** [review]
Overall, it's good to me. Please add the comment suggested from Robert.

File: tests/functional/test_charm.py (hunk 1 above)
Comment (jneo8) [reply tests/functional/test_charm.py:17]: The upstream of pytest-operator provides the fix to it 12 hours ago. This line is not needed now.

https://github.com/charmed-kubernetes/pytest-operator/pull/120

**Review by jneo8 — CHANGES_REQUESTED** [review]
LGTM. Just need to remove the line which provide pytest-asyncio fix.

**Comment by chanchiwai-ray** [comment]
Not related to this PR, are we actually perform the static test of the alert rules?

File: tests/functional/test_charm.py (hunk 1 above)
Comment (dashmage) [reply tests/functional/test_charm.py:17]: I've confirmed what @jneo8 has mentioned and with the version pin on `pytest-asyncio`, things look good for us. I'll remove this line and push the change.

**Comment by dashmage** [comment]
> Not related to this PR, are we actually performing the static test of the alert rules?

If you mean running `promtool`, this is done manually before committing the code currently. There's no automated process in place yet :(

**Comment by dashmage** [comment]
Update: Functional tests are green, looks good to merge!
[Latest job run](https://github.com/canonical/hardware-observer-operator/actions/runs/7097409972/job/19317515361?pr=121)

**Review by rgildein — APPROVED** [review]
LGTM

**Review by jneo8 — APPROVED** [review]

**Pull request merged**